lint/rule_template_datasource.go

```diff
@@ -2,47 +2,95 @@ package lint
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 )
 
+// This rule checks that the dashboard has single templated datasource
+func (d *Dashboard) checkPrometheusOrLokiDS(template *Template) Result {
+
+	if template.Name != "datasource" {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable named '%s', should be named 'datasource'", d.Title, template.Name),
+		}
+	}
+
+	if template.Label != "Data Source" {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable labeled '%s', should be labeled 'Data Source'", d.Title, template.Label),
+		}
+	}
+
+	if template.Query != Prometheus && template.Query != Loki {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable query is '%s', should be 'prometheus' or 'loki'", d.Title, template.Query),
+		}
+	}
```

Comment:
While the dashboard-linter is currently very focused on prometheus (and increasingly, loki) datasources, we should only raise errors for things that we can check, and for which an opinionated best practice exists.

In this case, it's perfectly valid to have a dashboard with a primary (or single) datasource which is neither prometheus nor loki. The dashboard-linter simply has no opinion or rule for those dashboards.